http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/model/LeoAuthProvider.scala

```diff
@@ -19,39 +19,44 @@ import org.broadinstitute.dsde.workbench.model.{TraceId, UserInfo, WorkbenchEmai
 
 // Typeclass representing a Sam resource and associated policies
 sealed trait SamResource[R] {
-  def resourceType: SamResourceType
+  def resourceType(r: R): SamResourceType
```

Comment:
By moving from val to def, we avoid having to pass around the accessScope parameter. It is way cleaner this way and make the implicits happy!